src/main/scala/nl.knaw.dans.easy.authinfo/Licenses.scala

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.authinfo
+
+import nl.knaw.dans.easy.authinfo.components.FileRights
+import nl.knaw.dans.easy.authinfo.components.RightsFor.ANONYMOUS
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.apache.commons.configuration.PropertiesConfiguration
+
+import scala.util.Try
+import scala.xml.Node
+
+class Licenses(licenses: PropertiesConfiguration) extends DebugEnhancedLogging {
+
+  private val CC_0_LICENSE = "http://creativecommons.org/publicdomain/zero/1.0"
+  private val DANS_LICENSE = "http://dans.knaw.nl/en/about/organisation-and-policy/legal-information/DANSGeneralconditionsofuseUKDEF.pdf"
+
+  def getLicense(dcmiMetadata: Option[Node], rights: FileRights): Try[License] = Try {
+    def getValue(tag: String): Option[String] = {
+      dcmiMetadata.flatMap(node => (node \ tag).headOption.map(_.text))
+    }
+
+    val licenseKey = getValue("license")
+    licenseKey.map(key => License(key, getLicenseTitle(key))).getOrElse(getDefaultLicense(rights))
```

Comment:
If there is no license given in `dataset.xml` then a default license will be added, and that is `CC0` for Open Access datasets and `DANS license` otherwise.